public/app/features/explore/LogRow.tsx

```diff
@@ -135,6 +136,8 @@ export class LogRow extends PureComponent<Props, State> {
     const highlightClassName = classnames('logs-row__match-highlight', {
       'logs-row__match-highlight--preview': previewHighlights,
     });
+    const containsAnsiCodes = hasAnsiCodes(row.entry);
```

Comment:
did you test the performance effect of adding this check? As it's a regex check that is always executed on all log lines

Reply:
Yes, it was <5%. But I havent document my method anywhere, so I might check again. Might make sense to do this check when processing the log entry and then keep a flag instead of doing this in each render.